src/seedu/addressbook/parser/Parser.java

```diff
@@ -390,6 +404,21 @@ private Command prepareFeesList() {
         }
     }
 
+    /**
+     * Parses arguments in the context of the view grades command.
+     *
+     * @param args full command args string
+     * @return the prepared command
+     */
+    private Command prepareViewGrades(String args) {
+        try {
+            final int targetIndex = parseArgsAsDisplayedIndex(args);
+            return new ViewGradesCommand(targetIndex);
+        } catch (ParseException | NumberFormatException e) {
+            return new IncorrectCommand(String.format(MESSAGE_INVALID_COMMAND_FORMAT,
+                    ViewGradesCommand.MESSAGE_USAGE));
+        }
+    }
 
```

Comment:
As I mentioned in the previous patch notes, to reduce the number of prepare methods we have, I recommend using prepareSingleIndexCommand(String, Command) instead for all commands with only a single index as arguments
Can refer to `case DeleteCommand.COMMAND_WORD:`